mlir/utils/jenkins/Jenkinsfile

```diff
@@ -97,6 +97,8 @@ pipeline {
                                   ln -s build/compile_commands.json compile_commands.json
                                 fi
                                 '''
+                                // Before premerge-checks, reset all mods from the OOT patcher tool
+                                sh 'git checkout external/llvm-project && git clean -fd external/llvm-project'
```

Comment:
[Question] Does this effectively disable premerge check on llvm repo changes?

Reply:
No, but it does disable premerge checks on our patches (for as long as we're using our current patch strategy)

Reply:
I'm fine with what we have now (to unblock everyone from a broken CI). But I'd prefer to a ticket that aims at re-enable `permerge-checks` later. Otherwise, it will be annoying when we want to upstream those changes (that requires us to format the code first).

Reply:
How are we going to upstream patches under this patching infrastructure? Shouldn't we add and commit the changes first? I think the patches that cmake create are those we don't want to upstream, so it's Ok to disable pre-merge checks for them. Am I right?

Reply:
Adding and committing the changes is the plan, but we went forward with the current patching infrastructure to get *something* working real quick.

Reply:
> How are we going to upstream patches to llvm under this patching infrastructure?

That's definitely a @parallelo question

> I think the patches that cmake create are those we don't want to upstream

It's the opposite. The patches are changes that we do want to upstream. They are patches only temporarily (before upstream is completed).

Reply:
My point was if we add and commit the patches first, they wouldn't skip the pre-merge checks. The pre-merge checks shouldn't worry about changes that are created during the build process.

Reply:
Having said that, I agree we need a different patching infrastructure.

Reply:
> The pre-merge checks shouldn't worry about changes that are created during the build process.

In the general sense, it shouldn't. In our particular case it should. Having a nu-formatted patch will make it harder to upstream our changes. It is just naturally a developer's duty to keep code formatted, even if it is in the patch repo. So far the best way to enforce that is to use premerge check on it as well.

Reply:
Just catching up here.  We are in agreement that the current patching approach should be temporary.  Upstreaming as much as possible is the way forward, and then we store our non-upstreamable stuff in the git-subtree itself.

Reply:
Side note - we can discuss upstreaming details in more depth here:  https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/231

Reply:
I've stored everything in the subtree for now in my latest PR.